samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -400,6 +349,7 @@ private void restoreStores() throws InterruptedException {
     });
 
     LOG.info("Store Restore complete");
+    return taskCheckpoints;
```

Comment:
Don't you need to return the new checkpoint here?

Reply:
The new checkpoints are reflected in the taskCheckpoints.

Reply:
Updated to now return a new map of taskname and checkpoint instead of mutating taskCheckpoints